map-utils.js

```diff
@@ -1,126 +1,157 @@
+  var endLng = degreesToRadians(posB.lng || posB.longitude);
+  var partialResult = Math.sin(startLat) * Math.sin(endLat) +
+    Math.cos(startLat) * Math.cos(endLat) * Math.cos(startLng - endLng);
+  var distance = Math.acos(Math.min(partialResult, 1)) * EARTH_RADIUS;
+
+  return Number(distance.toFixed(3));
+};
+var getRadiusLatLng = function (latLng, radius) {
+  'use strict';
+  var startLat = degreesToRadians(latLng.lat);
+  var startLng = degreesToRadians(latLng.lng);
+  var dist = parseFloat(radius) / (EARTH_RADIUS * 1000);
+  var destAng = degreesToRadians(0);
+  var destLat = Math.asin(Math.sin(startLat) * Math.cos(dist) +
+  	Math.cos(startLat) * Math.sin(dist) * Math.cos(destAng));
+  var destLng = ((startLng + Math.atan2(Math.sin(destAng) * Math.sin(dist) *
+  	Math.cos(startLat), Math.cos(dist) - Math.sin(startLat) *
+  	Math.sin(destLat))) * 180) / Math.PI;
+  destLat = (destLat * 180) / Math.PI;
+
+  return { lat: destLat, lng: destLng };
+};
+var getPointFromLatLng = function (latLng, opts) {
+  'use strict';
+  opts = opts || {asPlainObject: false};
+  var siny = Math.max(Math.sin(degreesToRadians(latLng.lat)), -0.9999);
+  var x = TILE_SIZE * (0.5 + latLng.lng / 360);
+  var y = TILE_SIZE * (0.5 - Math.log((1 + siny) / (1 - siny)) / (4 * Math.PI));
+  if (opts.asPlainObject) {
+    return {x: x, y: y};
+  };
+
+  return new x, y;
+};
 var getLatLngFromPoint = function (point) {
-var lng = (point.x - tilePixelCenter) / pixelsPerLngDegree;
-var latRadians = (point.y - tilePixelCenter) / -pixelsPerLngRadian; 
-var lat = radiansToDegrees(2 * Math.atan(Math.exp(latRadians)) - Math.PI / 2);
-return {lat: lat, lng: lng} 
-}
+  'use strict';
+  var lng = (point.x - tilePixelCenter) / pixelsPerLngDegree;
+  var latRadians = (point.y - tilePixelCenter) / -pixelsPerLngRadian;
+  var lat = radiansToDegrees(2 * Math.atan(Math.exp(latRadians)) - Math.PI / 2);
+
+  return {lat: lat, lng: lng};
+};
 var getFormattedPlaceName = function (placeObj) {
-if (!placeObj.formatted_address) {
-return null
-}
-var htmlAddress = placeObj.adr_address
-var firstPart = htmlAddress.substr(0, htmlAddress.indexOf("<"))
-var addressContainer = window.document.createElement("span")
-addressContainer.innerHTML = htmlAddress
-var childNodes = Array.prototype.slice.call(addressContainer.children, 0)
-if (childNodes.length == 0) {
-return placeObj.formatted_address
-}
-return firstPart + childNodes
-  .filter(function (node) { return node.className != "postal-code"})
-  .map(function (node) { return node.textContent})
-  .join(", ")  
-}
+  'use strict';
+  if (!placeObj.formatted_address) {
+    return null;
+  };
+
+  var htmlAddress = placeObj.adr_address;
+  var firstPart = htmlAddress.substr(0, htmlAddress.indexOf('<'));
+  var addressContainer = window.document.createElement('span');
+  addressContainer.innerHTML = htmlAddress;
+  var childNodes = Array.prototype.slice.call(addressContainer.children, 0);
+  if (childNodes.length == 0) {
```

Comment:
Upon closer inspection, it looks like you're missing a rule required in the assignment:

* always use strict equality/inequality*